src/opendr/perception/pose_estimation/hr_pose_estimation/high_resolution_learner.py

```diff
@@ -0,0 +1,610 @@
+            if kernel > 0:
+                pool_img = HighResolutionPoseEstimationLearner.__pooling(self, img, kernel)
+
+            else:
+                pool_img = img
+
+            # ------- Heatmap Generation -------
+            avg_pafs = HighResolutionPoseEstimationLearner.__first_pass(self, self.model, pool_img)
+            avg_pafs = avg_pafs.astype(np.float32)
+
+            pafs_map = cv2.blur(avg_pafs, (5, 5))
+            pafs_map[pafs_map < self.threshold] = 0
+
+            heatmap = pafs_map.sum(axis=2)
+            heatmap = heatmap * 100
+            heatmap = heatmap.astype(np.uint8)
+            heatmap = cv2.blur(heatmap, (5, 5))
+
+            contours, hierarchy = cv2.findContours(heatmap, cv2.RETR_TREE, cv2.CHAIN_APPROX_SIMPLE)
+            count = []
+            coco_keypoints = []
+
+            if len(contours) > 0:
+                for x in contours:
+                    count.append(x)
+                xdim = []
+                ydim = []
+
+                for j in range(len(count)):  # Loop for every human (every contour)
+                    for i in range(len(count[j])):
+                        xdim.append(count[j][i][0][0])
+                        ydim.append(count[j][i][0][1])
+
+                h, w, _ = pool_img.shape
+                xmin = int(np.floor(min(xdim))) * int((w / heatmap.shape[1])) * kernel
+                xmax = int(np.floor(max(xdim))) * int((w / heatmap.shape[1])) * kernel
+                ymin = int(np.floor(min(ydim))) * int((h / heatmap.shape[0])) * kernel
+                ymax = int(np.floor(max(ydim))) * int((h / heatmap.shape[0])) * kernel
+
+                extra_pad_x = int(self.perc * (xmax - xmin))  # Adding an extra pad around cropped image
+                extra_pad_y = int(self.perc * (ymax - ymin))
+
+                if xmin - extra_pad_x > 0:
+                    xmin = xmin - extra_pad_x
+                if xmax + extra_pad_x < img.shape[1]:
+                    xmax = xmax + extra_pad_x
+                if ymin - extra_pad_y > 0:
+                    ymin = ymin - extra_pad_y
+                if ymax + extra_pad_y < img.shape[0]:
+                    ymax = ymax + extra_pad_y
+
+                if (xmax - xmin) > 40 and (ymax - ymin) > 40:
+                    crop_img = img[ymin:ymax, xmin:xmax]
+                else:
+                    crop_img = img[offset:img.shape[0], offset:img.shape[1]]
+
+                h, w, _ = crop_img.shape
+
+                # ------- Second pass of the image, inference for pose estimation -------
+                avg_heatmaps, avg_pafs, scale, pad = \
+                    HighResolutionPoseEstimationLearner.__second_pass(self,
+                                                                      self.model, crop_img,
+                                                                      self.second_pass_height, max_width,
+                                                                      self.stride, upsample_ratio)
```

Comment:
Same as previous comments for accessing the private method, same comment for passing the model.
```suggestion
                avg_heatmaps, avg_pafs, scale, pad = self.__second_pass(self.model, crop_img,
                                                                        self.second_pass_height, max_width,
                                                                        self.stride, upsample_ratio)
```